TestMapping.test_keys fails under MPI

* Operating System: Fedora rawhide

```
+ mpirun /usr/bin/python3 -m pytest --pyargs h5py -rxXs --with-mpi
============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.0.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/orion/fedora/h5py/h5py-2.10.0
plugins: cov-2.10.1, arraydiff-0.3, doctestplus-0.7.0, pyfakefs-3.5.8, mpi-0+unknown, remotedata-0.3.2, openfiles-0.5.0, mock-3.3.1, xdist-2.1.0, timeout-1.4.2, forked-1.3.0
```
```
=================================== FAILURES ===================================
____________________________ TestMapping.test_keys _____________________________

self = <h5py.tests.test_base.TestMapping testMethod=test_keys>

...                            [ 96%]
tests/test_vds/test_lowlevel_vds.py     def setUp(self):
        data = ('a', 'b')
>       self.f = File('foo.hdf5', 'w')

../../../rpmbuild/BUILDROOT/h5py-2.10.0-4.git.fc34.x86_64/usr/lib64/python3.9/site-packages/openmpi/h5py/tests/test_base.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../rpmbuild/BUILDROOT/h5py-2.10.0-4.git.fc34.x86_64/usr/lib64/python3.9/site-packages/openmpi/h5py/_hl/files.py:424: in __init__
    fid = make_fid(name, mode, userblock_size,
../../../rpmbuild/BUILDROOT/h5py-2.10.0-4.git.fc34.x86_64/usr/lib64/python3.9/site-packages/openmpi/h5py/_hl/files.py:196: in make_fid
    fid = h5f.create(name, h5f.ACC_TRUNC, fapl=fapl, fcpl=fcpl)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OSError: Unable to create file (unable to lock file, errno = 11, error message = 'Resource temporarily unavailable')

h5py/h5f.pyx:116: OSError
=============================== MPI Information ================================
rank: 1
size: 4
MPI version: 3.1
MPI library version: Open MPI v4.0.5, package: Open MPI mockbuild@buildvm-x86-11.iad2.fedoraproject.org Distribution, ident: 4.0.5, repo rev: v4.0.5, Aug 26, 2020
MPI vendor: Open MPI 4.0.4
mpi4py rc: 
 initialize: True
 threads: True
 thread_level: multiple
 finalize: None
 fast_reduce: True
 recv_mprobe: True
 errors: exception
mpi4py config:
 mpicc: /usr/lib64/openmpi/bin/mpicc
 mpicxx: /usr/lib64/openmpi/bin/mpicxx
 mpifort: /usr/lib64/openmpi/bin/mpifort
 mpif90: /usr/lib64/openmpi/bin/mpif90
 mpif77: /usr/lib64/openmpi/bin/mpif77
```

--- a/h5py/tests/test_base.py
+++ b/h5py/tests/test_base.py
@@ -69,8 +69,8 @@ class TestMapping(BaseTest):
     """
 
     def setUp(self):
+        super().setUp()
         data = ('a', 'b')
-        self.f = File('foo.hdf5', 'w')
         self.grp = self.f.create_group('bar')
         self.attr = self.f.attrs.create('x', data)
 

--- a/h5py/tests/test_base.py
+++ b/h5py/tests/test_base.py
@@ -74,10 +74,6 @@ class TestMapping(BaseTest):
         self.grp = self.f.create_group('bar')
         self.attr = self.f.attrs.create('x', data)
 
-    def TearDown(self):
-        if self.f:
-            self.close()
-
     def test_keys(self):
         key_1 = self.f.keys()
         self.assertIsInstance(repr(key_1), str)

